### PR TITLE
Adding new option to skip filters calling BLAT

### PR DESCRIPTION
**Pull request by athos**
To allow users to use fusionfusion without depending on BLAT (which has a license-related difficulty for commercial use), this PR adds a new option to skip the filters that call BLAT.

This change makes use of the similarity of the file formats between `*.chimeric.clustered.splicing.txt` and `*.fusion.result.txt`, and mix and match a couple of column data from the former.

### CODE REVIEW

File: fusionfusion/annotationFunction.py [hunk 1]
```diff
@@ -190,8 +190,12 @@ def filterAndAnnotation(inputFilePath, outputFilePath, genome_id, is_grc):
...
         if filter_same_gene == True and sameGeneFlag == 1: continue
 
-        print('\t'.join(F[0:8]) + '\t' + ';'.join(gene1) + '\t' + ';'.join(junction1) + '\t' + ';'.join(gene2) + '\t' + ';'.join(junction2) + '\t' + \
-              F[11] + '\t' + F[12] + '\t' + F[16] + '\t' + F[17], file = hOUT)
+        line = '\t'.join(F[0:8]) + '\t' \
+            + ';'.join(gene1) + '\t' + ';'.join(junction1) + '\t' \
+            + ';'.join(gene2) + '\t' + ';'.join(junction2)
+        if includes_scores:
+            line += '\t' + F[11] + '\t' + F[12] + '\t' + F[16] + '\t' + F[17]
+        print(line, file = hOUT)
```
Comment (alumi) [inline fusionfusion/annotationFunction.py]: https://www.python.org/dev/peps/pep-0008/#other-recommendations

```suggestion
        print(line, file=hOUT)
```

File: fusionfusion/annotationFunction.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def get_junc_info(chr, pos, ref_exon_tb, ens_exon_tb, junction_margin):
 
 
 
-def filterAndAnnotation(inputFilePath, outputFilePath, genome_id, is_grc):
+def filterAndAnnotation(inputFilePath, outputFilePath, genome_id, is_grc, includes_scores=True):
```
Comment (alumi) [inline fusionfusion/annotationFunction.py]: https://www.python.org/dev/peps/pep-0008/#maximum-line-length

```suggestion
def filterAndAnnotation(inputFilePath,
                        outputFilePath,
                        genome_id,
                        is_grc,
                        includes_scores=True):
```

File: fusionfusion/arg_parser.py [hunk 3]
```diff
@@ -40,6 +40,9 @@ def create_parser():
     parser.add_argument("--pooled_control_file", default = None, type = str,
                         help = "the path to control data created by merge_control (default: %(default)s)")
 
+    parser.add_argument("--no_blat", default = False, action = 'store_true',
+                        help = "if true, skip filters using BLAT")
```
Comment (alumi) [inline fusionfusion/arg_parser.py]:
```suggestion
    parser.add_argument("--no_blat", default=False, action='store_true',
                        help="if true, skip filters using BLAT")
```

File: fusionfusion/run.py [hunk 4]
```diff
@@ -33,42 +33,52 @@ def cluster_filter_junction(inputFilePath, outputFilePrefix, args):
...
-                                         outputFilePrefix + ".chimeric.clustered.filt3.txt",
-                                         outputFilePrefix + ".chimeric.clustered.splicing.contig.check.txt")
-    
-        annotationFunction.filterAndAnnotation(outputFilePrefix + ".chimeric.clustered.filt3.txt",
-                                         outputFilePrefix + ".fusion.result.txt", args.genome_id, args.grc)
-
+    if args.no_blat:
+        annotationFunction.filterAndAnnotation(outputFilePrefix + ".chimeric.clustered.splicing.txt",
+                                               outputFilePrefix + ".fusion.result.txt",
+                                               args.genome_id, args.grc, includes_scores=False)
```
Comment (alumi) [inline fusionfusion/run.py]:
```suggestion
        annotationFunction.filterAndAnnotation(
            outputFilePrefix + ".chimeric.clustered.splicing.txt",
            outputFilePrefix + ".fusion.result.txt",
            args.genome_id, args.grc, includes_scores=False)
```

**Review by alumi — APPROVED** [review]
Thank you for working on this! :+1: The logic looks good to me. I've added some comments on coding styles.

**Comment by athos** [comment]
@alumi Thanks for the suggestions! I just applied them all 👍

I'm going to squash the commits into one when rebasing them onto the `devel` branch after the PR is approved.

**Review by alumi — APPROVED** [review]
LGTM 👍 👍

File: fusionfusion/annotationFunction.py [hunk 5]
```diff
@@ -222,23 +230,23 @@ def merge_fusion_result(input_dir, output_file_path):
         with open(input_dir + "/ms2.fusion.result.txt") as hIN:
             for line in hIN:
                 F = line.rstrip('\n').split('\t')
-                key = '\t'.join(F[0:7] + F[8:12])
+                key = '\t'.join(F[0:7] + F[8:])
```
Comment (xckitahara) [inline fusionfusion/annotationFunction.py]: This change seems to affect the traditional case, but is it intended?

Comment (athos) [reply fusionfusion/annotationFunction.py]: I intentionally omitted those at the time, but after rechecking the code, the changes around here indeed look needless now 🤔 
Fortunately, they don’t seem to do any real harm unless we specify more than one of the three options `--star`, `--ms2` and `--th2` at the same time, but I’ll revert them either way.
Thank you for pointing this out!

Comment (athos) [reply fusionfusion/annotationFunction.py]: I've reverted these portions of the change at f05ea5b 👍  Also, I’ve reconfirmed that the result is identical to before.

**Review by xckitahara — APPROVED** [review]
LGTM 👍

**Review by xckitahara — APPROVED** [review]

**Comment by athos** [comment]
@alumi @xckitahara Thank you for reviewing! I'm moving on to preparing a new PR for the upstream 💪 

I will close this PR, but I think it's better to leave the `feature/no-blat` branch as-is for a while since it would be useful for the review for the upcoming feature (`feature/short-range-chimera-filter`).

**Pull request closed without merging**